src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -1214,6 +1228,26 @@ def create_oracledb_connection():
         struct_schema = partitioner.schema
         partitioned_queries = partitioner.partitions
 
+        if udtf_configs is not None:
+            if "external_access_integration" not in udtf_configs:
+                raise ValueError(
+                    "external_access_integration cannot be None when udtf ingestion is used."
+                )
+            partitions_table = random_name_for_temp_object(TempObjectType.TABLE)
+            self._session.create_dataframe(
+                [[query] for query in partitioned_queries], schema=["partition"]
+            ).write.save_as_table(partitions_table, table_type="temp")
+            df = partitioner.driver.udtf_ingestion(
+                self._session,
+                struct_schema,
+                partitions_table,
+                udtf_configs["external_access_integration"],
+                fetch_size=fetch_size,
+                imports=udtf_configs.get("imports", None),
+                packages=udtf_configs.get("packages", None),
+            )
+            return df
```

Comment:
before returning the DF, we need to set `set_api_call_source(res_df, DATA_SOURCE_DBAPI_SIGNATURE)` to track the source of this DF for telemetry purpose.

check the existing returning code